tests/testing/make_sdists.py

```diff
@@ -6,12 +6,13 @@
 def random_string():
     """return a short suffix that shouldn't collide with any subsequent calls"""
     import os
-    import base64
+    import string
+    import random
 
-    return '.'.join((
-        str(os.getpid()),
-        base64.urlsafe_b64encode(os.urandom(3)),
-    ))
+    return '{}.{}'.format(
+        os.getpid(),
+        ''.join(random.choice(string.ascii_lowercase) for x in range(8)),
```

Comment:
Mm hmm.

the solution is `.decode('US-ASCII')`

On Wed, Dec 9, 2015 at 5:44 PM Chris Kuehl <notifications@github.com> wrote:

> In tests/testing/make_sdists.py
> <https://github.com/bukzor/pip-faster/pull/13#discussion_r47180045>:
>
> >
> > -    return '.'.join((
> > -        str(os.getpid()),
> > -        base64.urlsafe_b64encode(os.urandom(3)),
> > -    ))
> > +    return '{}.{}'.format(
> > +        os.getpid(),
> > +        ''.join(random.choice(string.ascii_lowercase) for x in range(8)),
>
> The problem is base64.urlsafe_b64encode wants a string, not bytes, on
> Python 3. You'd have to use base64.encodebytes which isn't Python
> 2-compatible. I figured switching to this was better than using version
> detection.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bukzor/pip-faster/pull/13/files#r47180045>.
>